File: src/agent/tools/builtin/web-search.ts
import type { ToolHandler } from "../../../types";

export const webSearchTool: ToolHandler = {
  definition: {
    name: "web_search",
    description: "Search the web for current information. Returns search results with titles, snippets, and URLs.",
    parameters: {
      type: "object",
      properties: {
        query: {
          type: "string",
          description: "The search query",
        },
      },
      required: ["query"],
    },
  },

  async execute(args) {
    const query = args.query as string;
    // Placeholder: integrate with a real search API (SerpAPI, Tavily, etc.)
    return JSON.stringify({
      note: "Web search is not yet configured. Please set up a search API provider.",
      query,
      results: [],
    });
  },
};

File: src/tools/builtins/web_search.py
"""Web search tool — placeholder, matching TS tools/builtin/web-search.ts."""

from __future__ import annotations

import json

from src.types import ToolDefinition, ToolHandler


async def _execute(args: dict) -> str:
    query = args.get("query", "")
    return json.dumps({
        "note": "Web search is not yet implemented. Integration needed with SerpAPI, Tavily, or similar.",
        "query": query,
        "results": [],
    })


web_search_tool = ToolHandler(
    definition=ToolDefinition(
        name="web_search",
        description="Search the web for information. (Not yet implemented — placeholder.)",
        parameters={
            "type": "object",
            "properties": {
                "query": {
                    "type": "string",
                    "description": "The search query.",
                },
            },
            "required": ["query"],
        },
    ),
    execute=_execute,
)
